go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -17,96 +36,159 @@ def initialize(dependencies:, go_mod:, go_sum:, credentials:)
         end
 
         def updated_go_mod_content
```

Comment:
It turned out that updating the `go.mod` and the `go.sum` required some common bits and pieces (such as running `go get`), so I've merged the implementations, and the original methods just reference the results.